extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -1542,3 +1372,52 @@ export class AmpA4A extends AMP.BaseElement {
    */
   emitLifecycleEvent(unusedEventName, opt_extraVariables) {}
 }
+
+const SIGNATURE_FORMAT = new RegExp(
+    '^([A-Za-z0-9._-]+):([A-Za-z0-9._-]+):' +
+    '((?:[A-Za-z0-9+/]{4})*(?:[A-Za-z0-9+/]{2}[A-Za-z0-9+/=]=)?)$');
+
+/**
+ * Decode the `AMP-Fast-Fetch-Signature` header, in the format
+ * `{signingServiceName}:{keypairId}:{base64Signature}`, to a signature info
+ * object.
+ *
+ * @param {?string} headerValue
+ * @return {?SignatureInfoDef}
+ */
+export function decodeSignatureHeader(headerValue) {
+  if (!headerValue) {
+    return null;
+  }
+  const match = SIGNATURE_FORMAT.exec(headerValue);
+  if (!match) {
+    // TODO(@taymonbeal, #9274): replace this with real error reporting
+    user().error(TAG, `Invalid signature header: ${headerValue}`);
+    return null;
+  }
+  return /** @type {?SignatureInfoDef} */ ({
+    signingServiceName: match[1],
+    keypairId: match[2],
+    signature: base64DecodeToBytes(match[3]),
```

Comment:
Not extremely familiar with the codebase, but would errors in decoding surface here implicitly?